internal/datastores/nfs_datastore_subresource.go

```diff
@@ -0,0 +1,91 @@
+/* Copyright 2023 VMware, Inc.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package datastores
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	validation_utils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+)
+
+// NfsDatastoreSchema this helper function extracts the NFS Datastore schema, so that
+// it's made available for both Domain and Cluster creation.
+func NfsDatastoreSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"datastore_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Datastore name used for cluster creation",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"path": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Shared directory path used for NFS based cluster creation",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"read_only": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Readonly is used to identify whether to mount the directory as readOnly or not",
+			},
+			"server_name": {
```

Comment:
There is probably a more appropriate name for this.

Reply:
This is what the API reference is using: https://developer.vmware.com/apis/vcf/4.5.0/data-structures/NasVolumeSpec/

I'm using it for consistency.